File: python/programmers/level 2/영어 끝말잇기.py
def solution(n, words):
    using = []
    turn = [1, 1]
    for word in words:
        if len(using) > 0 and using[-1][-1] != word[0]:
            return turn
        elif word in using:
            return turn
        else:
            using.append(word)
            turn[0] += 1
            if turn[0] // (n + 1) > 0 :
                turn[1] += 1
                turn[0] = turn[0] % n
    return [0, 0]

File: java/src/main/java/ps_traning/programmers/level_2/영어_끝말잇기.java
package ps_traning.programmers.level_2;

import java.util.HashSet;
import java.util.Set;

public class 영어_끝말잇기 {
    public int[] solution(int n, String[] words) {
        int[] answer = {0, 0};
        String before = words[0];
        Set<String> set = new HashSet<>();
        for (int i = 1; i < words.length; i++) {
            set.add(before);
            String w = words[i];
            if (before.charAt(before.length() - 1) != w.charAt(0) || set.contains(w)) {
                answer[0] = i % n + 1;
                answer[1] = i / n + 1;
                break;
            }
            before = w;
        }

        return answer;
    }
}
